benchmarks/wrappers.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+
+from .common import Benchmark
+
+
+class TimeWrappers(Benchmark):
+    params = [
+        "h5py.Dataset",
+        "InMemoryArrayDataset",
+        "InMemoryDataset",
+        "InMemorySparseDataset",
+    ]
+    param_names = ["kind"]
+
+    def setup(self, kind):
+        super().setup()
+        if kind == "h5py.Dataset":
+            self.file.create_dataset(
+                "data",
+                data=np.arange(10000).reshape((100, 10, 10)),
+                chunks=(3, 3, 3),
+                maxshape=(None, None, None),
+            )
+            self.reopen()
+            self.ds = self.file["data"]
+        elif kind == "InMemoryArrayDataset":
+            self.ctx = self.vfile.stage_version("v0")
+            version = self.ctx.__enter__()
+            self.ds = version.create_dataset(
+                "data",
+                data=np.arange(10000).reshape((100, 10, 10)),
+                chunks=(3, 3, 3),
+            )
+        elif kind == "InMemoryDataset":
+            with self.vfile.stage_version("v0") as ctx:
+                self.ds = ctx.create_dataset(
+                    "data",
+                    data=np.arange(10000).reshape((100, 10, 10)),
+                    chunks=(3, 3, 3),
+                    maxshape=(None, None, None),
+                )
+            self.reopen()
+            self.ctx = self.vfile.stage_version("v1")
+            version = self.ctx.__enter__()
+            self.ds = version["data"]
+        elif kind == "InMemorySparseDataset":
+            self.ctx = self.vfile.stage_version("v0")
+            version = self.ctx.__enter__()
+            self.ds = version.create_dataset(
+                "data",
+                shape=(100, 10, 10),
+                chunks=(3, 3, 3),
+                maxshape=(None, None, None),
+            )
+
+    def time_getattr(self, kind):
+        self.ds[:, 0, 0:6]
+
+    def time_setattr(self, kind):
+        self.assert_clean_setup()
+        self.ds[:, 0, 0:6] = -1
+        if kind != "h5py.Dataset":
+            self.ctx.__exit__(None, None, None)  # commit
+            del self.ctx
+
+    def time_resize_bigger(self, kind):
+        self.assert_clean_setup()
+        self.ds.resize((100, 100, 100))
+        if kind != "h5py.Dataset":
+            self.ctx.__exit__(None, None, None)  # commit
+            del self.ctx
+
+    def time_resize_smaller(self, kind):
+        self.assert_clean_setup()
+        self.ds.resize((10, 10, 10))
+        if kind != "h5py.Dataset":
+            self.ctx.__exit__(None, None, None)  # commit
+            del self.ctx
```

Comment:
Taken care of by `teardown`? Same comment in `time_setattr` and `time_resize_bigger`.

```suggestion
```

Reply:
The point is that I want to include the commit phase in the metrics. Otherwise it makes no sense to compare against pure h5py, which writes to disk within the same command. I'll add a comment about it.